lib/msf/util/host.rb

```diff
@@ -14,21 +14,18 @@ def self.normalize_host(host)
         return host if defined?(::Mdm) && host.kind_of?(::Mdm::Host)
         norm_host = nil
 
-        if (host.kind_of? String)
-
+        if host.kind_of?(String)
           if Rex::Socket.is_ipv4?(host)
-            # If it's an IPv4 addr with a port on the end, strip the port
-            if host =~ /((\d{1,3}\.){3}\d{1,3}):\d+/
-              norm_host = $1
-            else
-              norm_host = host
-            end
+            norm_host = host
           elsif Rex::Socket.is_ipv6?(host)
             # If it's an IPv6 addr, drop the scope
             address, scope = host.split('%', 2)
             norm_host = address
```

Comment:
Not to be a PITA, but r.e your above issue I also found this as well:

```
>> Rex::Socket.is_ipv6?('[::FFFF:129.144.52.38]')
=> false
>> Rex::Socket.is_ipv6?('::FFFF:129.144.52.38')
=> true
>> Rex::Socket.is_ipv6?('::FFFF:129.144.52.38:80')
=> false
>> Rex::Socket.is_ipv6?('[::FFFF:129.144.52.38]:80')
=> false
>> Rex::Socket.is_ipv6?('FEDC:BA98:7654:3210:FEDC:BA98:7654:3210')
=> true
```

According to https://www.ietf.org/rfc/rfc2732.txt this is valid syntax for an IPv6 address in a URL format.

Reply:
Also according to https://datatracker.ietf.org/doc/html/rfc4007 this is not actually the scope but rather the zone ID? We might want to update the comment accordingly.